src/platform/plugins/shared/workflows_management/public/features/workflow_execution_detail/ui/build_step_executions_tree.ts

```diff
@@ -82,17 +86,19 @@ export function buildStepExecutionsTree(
   const stepExecutionsMap: Map<string, WorkflowStepExecutionDto> = new Map();
   const computedPathsMap: Map<string, string[]> = new Map();
 
-  stepExecutions.forEach((stepExecution) => {
-    const computedPath = [
-      ...(stepExecution.scopeStack ? flattenStackFrames(stepExecution.scopeStack) : []),
-      stepExecution.stepId,
-    ];
-    const key = computedPath.join('>');
-    computedPathsMap.set(stepExecution.id!, computedPath);
-    stepExecutionsMap.set(key, {
-      ...stepExecution,
+  stepExecutions
+    .filter((stepExecution) => isVisibleStepType(stepExecution.stepType!))
```

Comment:
Why is the `stepType` optional? Shouldn't we check that it is defined?

```suggestion
    .filter((stepExecution) => stepExecution.stepType && isVisibleStepType(stepExecution.stepType))
```